pkg/shared/utils/sync/distributed_lock_mock.go

```diff
@@ -0,0 +1,39 @@
+package sync
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+var _ DistributedLock = &distributedLockMock{}
+
+// It appears that go-mocket does not offer a straightforward method to intercept the Rollback call. To make it easily
+// interceptable, this mock substitutes the rollback call in the Unlock method with a DELETE.
```

Comment:
https://github.com/Selvatico/go-mocket/blob/master/stmt.go#L195 does this help?

Reply:
Not sure. The comment says it is used to simulate broken connections